### PR TITLE
Feature/loss split

### LINKED ISSUE #2: Feature: add metric to split out loss logging
Split out validation loss into separate losses for each dataset, e.g. have a separate MEPS loss and ARA loss for evaluation.
The idea is to alternate batches during the evaluation phase. We want to keep the ordering of the dates the same (shuffle = False) corresponding to how we would evaluate regular training. 
This will be useful to check during multi-domain training how performance on different datasets compares.

### PR DESCRIPTION
**Pull request by Sooph-knmi**
## Description

Split out validation loss into separate losses for each dataset, e.g. have a separate MEPS loss and ARA loss for evaluation.
The idea is to alternate batches during the evaluation phase. We keep the ordering of the dates the same (shuffle = False) corresponding to how we would evaluate regular training. In the dataset, we pass the same graph for each model instance.

Config setup is as follows:
```
training:
  validation_metrics:
  - _target_: anemoi.training.losses.mse.WeightedMSELoss
    scalars: []
    ignore_nans: True
  - _target_: anemoi.training.losses.multidomain_mse.MultiDomainMSELoss
    loss_name: "ARA"
    ignore_nans: True
  - _target_: anemoi.training.losses.multidomain_mse.MultiDomainMSELoss
    loss_name: "MEPS"
    ignore_nans: True
```

Implemented output mask as a dict of output masks 
This PR also fixes a bug in the graph generation (the old graphs were not being overwritten)

## Type of Change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Issue Number
Closes #2 

## Additional Notes
Note: NaN values for the loss split are sent to GPU manually in the forecaster at the moment (`to_device`)
Note: the updated dataloader config uses ERA5 operational data for training/evaluating on the year 2023

### CODE REVIEW

**Comment by einrone** [comment]
Tested and reviewed. It has also been discussed to use "previous loss value for instead of NaN", this is a separate PR.
